src/testing/reduced_order.cpp

```diff
@@ -1,191 +1,72 @@
-    pcout << "dg reduced_order-petrov-galerkin created" <<std::endl;
-    dg_pod_petrov_galerkin->allocate_system ();
-
-    dealii::VectorTools::interpolate(dg_pod_petrov_galerkin->dof_handler,initial_condition,dg_pod_petrov_galerkin->solution);
-
-    pcout << "Create POD-Petrov-Galerkin ODE solver" << std::endl;
-    // Create ODE solver using the factory and providing the DG object
+    std::unique_ptr<FlowSolver::FlowSolver<dim,nstate>> flow_solver_petrov_galerkin = FlowSolver::FlowSolverFactory<dim,nstate>::select_flow_case(all_parameters, parameter_handler);
     ode_solver_type = Parameters::ODESolverParam::ODESolverEnum::pod_petrov_galerkin_solver;
-    std::shared_ptr<PHiLiP::ODE::ODESolverBase<dim, double>> ode_solver_petrov_galerkin = PHiLiP::ODE::ODESolverFactory<dim, double>::create_ODESolver_manual(ode_solver_type, dg_pod_petrov_galerkin, pod);
-
-    /*%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%*/
-    /*Time-averaged relative error, E = 1/n_t * sum_{n=1}^{n_t} (||U_FOM(t^{n}) - U_ROM(t^{n})||_L2 / ||U_FOM(t^{n})||_L2 )
-     *Refer to section 6.1 in "The GNAT method for nonlinear model reduction: Effective implementation and application to computational ﬂuid dynamics and turbulent ﬂows"
-     *Authors: Kevin Carlberg, Charbel Farhat, ulien Cortial,  David Amsallem
-     *Journal of Computational Physics, 2013
-     */
-
-    const unsigned int number_of_time_steps = static_cast<int>(ceil(finalTime/param.ode_solver_param.initial_time_step));
-    const double constant_time_step = finalTime/number_of_time_steps;
-
-    pcout << " Advancing solution by " << finalTime << " time units, using "
-          << number_of_time_steps << " iterations of size dt=" << constant_time_step << " ... " << std::endl;
-
-    ode_solver_implicit->allocate_ode_system();
-    ode_solver_galerkin->allocate_ode_system();
-    ode_solver_petrov_galerkin->allocate_ode_system();
-
-    double galerkin_error_norm_sum = 0;
-    double petrov_galerkin_error_norm_sum = 0;
+    std::shared_ptr<ProperOrthogonalDecomposition::OfflinePOD<dim>> pod_petrov_galerkin = std::make_shared<ProperOrthogonalDecomposition::OfflinePOD<dim>>(flow_solver_petrov_galerkin->dg);
+    flow_solver_petrov_galerkin->ode_solver =  PHiLiP::ODE::ODESolverFactory<dim, double>::create_ODESolver_manual(ode_solver_type, flow_solver_petrov_galerkin->dg, pod_petrov_galerkin);
+    flow_solver_petrov_galerkin->ode_solver->allocate_ode_system();
+    auto functional_petrov_galerkin = FunctionalFactory<dim,nstate,double>::create_Functional(all_parameters->functional_param, flow_solver_petrov_galerkin->dg);
 
-    unsigned int current_iteration = 0;
+    flow_solver_implicit->ode_solver->steady_state();
+    flow_solver_galerkin->ode_solver->steady_state();
+    flow_solver_petrov_galerkin->ode_solver->steady_state();
```

Comment:
why not just call `flow_solver_petrov_galerkin->run();`? (minus creating that functional)

Reply:
and set the `ode_solver_type` by passing in a modified parameter object to `create_FlowSolver` like you did in PR #134

Reply:
I have changed it to call `flow_solver_implicit->run()`. However, the reduced-order solvers do not support polynomial ramping and do not yet seem to need this anyway, therefore they should only ever call steady_state, so I will leave those as is. 
For the `ode_solver_type`, I think it is just faster and prettier to swap the ODE object for the one that I need, rather than add a function with a bunch of if statements to match an enum type that I specified.

Reply:
If they don't support polynomial_ramping then we would want to change `if(flow_solver_param.steady_state_polynomial_ramping){ ... }` to `if(flow_solver_param.steady_state_polynomial_ramping && 
     ((ode_solver_type != ODEEnum::pod_galerkin_solver) 
      && (ode_solver_type != ODEEnum::pod_galerkin_solver))){ ... }`

so that a user doesnt mistakenly try doing polynomial ramping with a POD-type ODE solver

Reply:
then would you be able to call `->run` for the other two `flow_solver` objects

Reply:
I added the check so a user doesnt mistakenly try doing polynomial ramping with a POD ODE solver. However, I still can't call run for the other flow_solver objects. I would still need to create a function with some if statements to match the solver type to the enum. Otherwise, they will all run implicit since that is what's in the parameter file and what the variable `ode_solver_type` is set to.

Reply:
You could just pass a modified copy of the parameters object to `create_FlowSolver()`

Reply:
I could, but its extra code for nothing if I can just call steady_state directly. Also, this won't be a problem anymore once we introduce the different flow solver types as we discussed.